### DISCUSSION TITLE
Why this code not excute like I want?

### DISCUSSION THREAD

**Discussion by zmm2tysu** [pmndrs/valtio#913] [Q&A] [answered]
```
const GlobalStore = proxy({
 data: {
    obj1: {
      text: 'hello',
    },
  },
})

export default ()  => {
   //not work
   const state = useProxy(GlobalStore.data.obj1);
   //work
   const state = useProxy(GlobalStore.data).obj1;

   return (
      <div>
         <span>{state.text}<span>
         <button onClick={() => {
            //not work
            GlobalStore.data = {obj1: {text: 'world'}}
            //work
            GlobalStore.data.obj1 = {text: 'world'}
         }}>update</button>
      </div>
   )
}
```
![image](https://github.com/pmndrs/valtio/assets/20849786/56a16258-a7a1-4625-bbb1-03c941d3255d)
the redux tools record globalStore is changed, but UI is not rerender.

Hi,  I want to know why this dosent work when I click the update button.If there is some concept I have missed or mixed, pls let me know, thx very much!

**Comment by dai-shi** [comment, marked as answer]
```js
const state = useProxy(GlobalStore.data.obj1);
// 👆 means like this 👇 
const foo = GlobalStore.data.obj1;
const state = useProxy(foo);

// When you do this:
GlobalStore.data = {obj1: {text: 'world'}}
// `foo` doesn't change

**Reply by zmm2tysu** [reply]
Thanks, I understantd that, but new confusion arises.

> ```js
> const state = useProxy(GlobalStore.data);
> 
> // this doesn't rerender either.
> GlobalStore.data = {obj1: {text: 'world'}}
> // this can work
> GlobalStore.data.obj1 = {text: 'world'}
> ```

Or, did you mean which part wrapped by useProxy is always constant, and ressign it won't trigger rerendering UI. Am I understand it correctly？

**Reply by dai-shi** [reply]
Yes. You need to subscribe to GlobalState itself.

**Reply by zmm2tysu** [reply]
Got ya, that really helps me a lot!
